File: src/main/java/bot/boobbot/flight/Command.java
package bot.boobbot.flight;

public interface Command {

    public void execute(Context ctx);

    default public String getName() {
        return this.getClass().getSimpleName().toLowerCase();
    }

    default public CommandProperties getProperties() {
        return this.getClass().getAnnotation(CommandProperties.class);
    }

}

File: src/main/java/bot/boobbot/flight/Command.kt
package bot.boobbot.flight

interface Command {

    val name: String
        get() = this.javaClass.simpleName.toLowerCase()

    val properties: CommandProperties
        get() = this.javaClass.getAnnotation(CommandProperties::class.java)

    val hasProperties: Boolean
        get() = this.javaClass.isAnnotationPresent(CommandProperties::class.java)

    fun execute(ctx: Context)

}
